src/openforms/tests/e2e/input_validation_base.py

```diff
@@ -159,4 +159,6 @@ def _assertBackendValidation(self, form: Form, key: str, value: JSONValue):
                 invalid_params = response.json()["invalidParams"]
                 names = [param["name"] for param in invalid_params]
                 expected_name = f"data.{key}"
-                self.assertIn(expected_name, names)
+                # For complex components (i.e. fileuploads), the error can
+                # be a nested key, so we check for `.startswith`:
+                self.assertTrue(any(name.startswith(expected_name) for name in names))
```

Comment:
this has collissions if you have keys like `foo` and `foobar` - the `foobar` component would show up while you're testing the `foo` component and that's a false positive.

I'd check that either the exact key is present or, or that `name.startswith(f"{expected_name}.")` to get that path separator in.

Reply:
I'd also like to opt in to this behaviour, the majority of components have exact names, so a classvar `fuzzy_match_invalid_param_names = True` as an override would work nicely.